tests/components/dnsip/__init__.py

```diff
@@ -26,7 +26,16 @@ async def query(self, hostname, qtype) -> dict[str, str]:
         """Return information."""
         if self.error:
             raise self.error
-        return [QueryResult]
+        if qtype == "AAAA":
+            results = [QueryResult(), QueryResult(), QueryResult(), QueryResult()]
+            results[0].host = "2001:db8:77::face:b00c"
+            results[1].host = "2001:db8:77::dead:beef"
+            results[2].host = "2001:db8::77:dead:beef"
+            results[3].host = "2001:db8:66::dead:beef"
+        else:
+            results = [QueryResult(), QueryResult()]
+            results[1].host = "1.1.1.1"
```

Comment:
Seems we're not using the `nameservers` in init so I guess you can move the results[0] here too to make it consistent.

Reply:
I've updated the QueryResult class so we can initialise it with an IP which should be cleaner.